src/sensors/mod.rs

```diff
@@ -293,24 +350,60 @@ impl Topology {
     /// Gets currently running processes (as procfs::Process instances) and stores
     /// them in self.proc_tracker
     fn refresh_procs(&mut self) {
-        //! current_procs is the up to date list of processus running on the host
-        let current_procs = process::all_processes().unwrap();
-
-        for p in current_procs {
-            let pid = p.pid;
-            let res = self.proc_tracker.add_process_record(p);
-            match res {
-                Ok(_) => {}
-                Err(msg) => panic!("Failed to track process with pid {} !\nGot: {}", pid, msg),
+        #[cfg(target_os = "linux")]
+        {
+            //current_procs is the up to date list of processus running on the host
+            if let Ok(procs) = process::all_processes() {
+                info!("Before refresh procs init.");
+                let current_procs = procs
+                    .iter()
+                    .map(IProcess::from_linux_process)
+                    .collect::<Vec<_>>();
+
+                info!("Before refresh procs loop.");
+                for p in current_procs {
+                    let pid = p.pid;
+                    let res = self.proc_tracker.add_process_record(p);
+                    match res {
+                        Ok(_) => {}
+                        Err(msg) => {
+                            panic!("Failed to track process with pid {} !\nGot: {}", pid, msg)
+                        }
+                    }
+                }
+            }
+        }
+        #[cfg(target_os = "windows")]
+        {
+            let pt = &mut self.proc_tracker;
+            pt.sysinfo.refresh_processes();
+            pt.sysinfo.refresh_cpu();
+            let current_procs = pt
+                .sysinfo
+                .processes()
+                .values()
+                .map(IProcess::from_windows_process)
+                .collect::<Vec<_>>();
+            for p in current_procs {
```

Comment:
idem above.

Reply:
I'll do the same as https://github.com/hubblo-org/scaphandre/pull/148/commits/e9c96cb59cc49529d72b2715fc37c4b7392e35be on windows, depending on your appreciation :)